File: web/src/main/kotlin/no/ssb/kostra/web/viewmodel/FileReportVm.kt
package no.ssb.kostra.web.viewmodel

import com.fasterxml.jackson.annotation.JsonInclude
import io.micronaut.serde.annotation.Serdeable
import no.ssb.kostra.validation.report.Severity

@Serdeable
@JsonInclude(JsonInclude.Include.ALWAYS)
data class FileReportVm(
    val innparametere: KostraFormVm,
    val antallKontroller: Int,
    val severity: Severity,
    val feil: Collection<FileReportEntryVm>
)

File: web-frontend/src/kostratypes/fileReportVm.ts
import KostraFormVm from "./kostraFormVm"
import FileReportEntryVm from "./fileReportEntryVm"
import KostraSeverity from "./kostraSeverity"
import Nullable from "./nullable"

interface FileReportVm {
    readonly innparametere: NonNullable<KostraFormVm>
    readonly antallKontroller: NonNullable<number>
    readonly severity: NonNullable<KostraSeverity>
    readonly feil: Nullable<FileReportEntryVm[]>
}

export default FileReportVm